server/neptune/workflows/internal/terraform/job/env_step_runner.go

```diff
@@ -1,25 +1,95 @@
 package job
 
 import (
-	"strings"
-
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities"
 	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/execute"
 	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/terraform"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/terraform/version"
+	"go.temporal.io/sdk/workflow"
 )
 
 type EnvStepRunner struct {
 	CmdStepRunner CmdStepRunner
 }
 
-func (e *EnvStepRunner) Run(executionContext *ExecutionContext, localRoot *terraform.LocalRoot, step execute.Step) (string, error) {
+func (e *EnvStepRunner) Run(ctx *ExecutionContext, localRoot *terraform.LocalRoot, step execute.Step) (EnvVar, error) {
 	if step.EnvVarValue != "" {
-		return step.EnvVarValue, nil
+		return NewEnvVarFromString(step.EnvVarName, step.EnvVarValue), nil
+	}
+
+	return e.getEnv(ctx, localRoot, step)
+}
+
+func (e *EnvStepRunner) getEnv(ctx *ExecutionContext, localRoot *terraform.LocalRoot, step execute.Step) (EnvVar, error) {
+	version := workflow.GetVersion(ctx, version.LazyLoadEnvVars, workflow.DefaultVersion, 1)
+
+	if version == workflow.DefaultVersion {
+		return e.getLegacyEnvVar(ctx, localRoot, step)
+	}
+
+	relPath := localRoot.RelativePathFromRepo()
+	envVars := map[string]string{
+		"BASE_REPO_NAME":  localRoot.Repo.Name,
+		"BASE_REPO_OWNER": localRoot.Repo.Owner,
+		"DIR":             ctx.Path,
+		"PROJECT_NAME":    localRoot.Root.Name,
+		"REPO_REL_DIR":    relPath,
+	}
+
+	return NewEnvVarFromCmd(step.EnvVarName, step.RunCommand, ctx.Path, envVars), nil
+}
+
+func (e *EnvStepRunner) getLegacyEnvVar(ctx *ExecutionContext, localRoot *terraform.LocalRoot, step execute.Step) (EnvVar, error) {
+	res, err := e.CmdStepRunner.Run(ctx, localRoot, step)
+	return NewEnvVarFromString(step.EnvVarName, res), err
+}
+
+type StringEnvVar struct {
+	name  string
+	value string
+}
+
+func (v StringEnvVar) ToActivityEnvVar() activities.EnvVar {
+	return activities.EnvVar{
+		Name:  v.name,
+		Value: v.value,
+	}
+}
+
+type EnvVar interface {
+	ToActivityEnvVar() activities.EnvVar
+}
+
+type CommandEnvVar struct {
+	name           string
+	command        string
+	dir            string
+	additionalEnvs map[string]string
+}
+
+func (v CommandEnvVar) ToActivityEnvVar() activities.EnvVar {
+	return activities.EnvVar{
+		Name: v.name,
+		Command: activities.StringCommand{
+			Command:        v.command,
+			Dir:            v.dir,
+			AdditionalEnvs: v.additionalEnvs,
+		},
```

Comment:
Maybe better to talk over this bits in person but some comments could help the reader more or at least when to use what.

Reply:
That's fair, this is basically translated from the yaml config so it depends on how someone expresses their env vars.